models/addresses_summary/optimism/addresses_summary_optimism_mainet.sql

```diff
@@ -0,0 +1,106 @@
+{{ config(
+    tags=['dunesql'],
+    alias = alias('mainet'),
+    materialized='incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['address'],
+    post_hook='{{ expose_spells(\'["optimism"]\',
+                                "sector",
+                                "addresses_summary",
+                                \'["Henrystats"]\') }}'
+    )
+}}
+
+{% if is_incremental() %}
+WITH 
+
+weekly_active_addresses as (
+    SELECT 
+        DISTINCT 
+            "from" as address 
+        FROM 
+        {{ source('optimism', 'transactions') }}
+        WHERE block_time >= date_trunc('day', now() - Interval '7' Day)
+)
+
+SELECT 
+    'optimism' as blockchain, 
+    wd.address, 
+    fa.block_time as first_active_time, 
+    fa.tx_hash as first_transaction_hash, 
+    SUM((l1_fee + (gas_used * gas_price))/1e18) as gas_spent,
+    MAX(ot.block_time) as last_active_time, 
+    CASE
+           WHEN date_diff('day', min(ot.block_time), max(ot.block_time)) > 1825 THEN '5 years old User'
+           WHEN date_diff('day', min(ot.block_time), max(ot.block_time)) > 1460 THEN '4 years old User'
+           WHEN date_diff('day', min(ot.block_time), max(ot.block_time)) > 1095 THEN '3 years old User'
+           WHEN date_diff('day', min(ot.block_time), max(ot.block_time)) > 730 THEN '2 years old User'
+           WHEN date_diff('day', min(ot.block_time), max(ot.block_time)) > 365 THEN '1 year old User'
+           WHEN date_diff('day', min(ot.block_time), max(ot.block_time)) > 91 THEN '3 months old User'
+           WHEN date_diff('day', min(ot.block_time), max(ot.block_time)) > 30 THEN '1 month old User'
+           WHEN date_diff('day', min(ot.block_time), max(ot.block_time)) > 7 THEN '1 week old User'
+           ELSE 'less than 1 week old User'
+    END as address_age, 
+    CASE
+            WHEN (COUNT(ot.hash)/(date_diff('day', min(ot.block_time), max(ot.block_time)) + 1)) >= 1 THEN 'Daily User'
+            WHEN (COUNT(ot.hash)/(date_diff('day', min(ot.block_time), max(ot.block_time)) + 1)) >= 0.142857142857 THEN 'Weekly User'
+            WHEN (COUNT(ot.hash)/(date_diff('day', min(ot.block_time), max(ot.block_time)) + 1)) >= 0.0333333333333 THEN 'Monthly User'
+            WHEN (COUNT(ot.hash)/(date_diff('day', min(ot.block_time), max(ot.block_time)) + 1)) >= 0.0027397260274 THEN 'Yearly User'
```

Comment:
number of transactions / number of days active (age)

soo for a daily user they need to  make at least one transaction per day 

for a weekly user they need to make at least one transaction per week which 1/7 = 0.142857.. same for monthly and yearly as well.

Reply:
I added + 1 because date_diff returns zero (causes an error since we're dividing by 0) if an address as only been active on one day... this is causing the label to be inaccurate when an address is only active one day so will fix that shortly.